src/fiber/network.rs

```diff
@@ -1203,7 +1204,21 @@ where
             NetworkActorCommand::ConnectPeer(addr) => {
                 // TODO: It is more than just dialing a peer. We need to exchange capabilities of the peer,
                 // e.g. whether the peer support some specific feature.
-                // TODO: If we are already connected to the peer, skip connecting.
+
+                if let Some(peer_id) = extract_peer_id(&addr) {
```

Comment:
We also need to check whether the address belongs to ourselves. Otherwise, we may encounter errors like `HandshakeError(SecioError(ConnectSelf))`.

Reply:
already checked in line 1213